appengine/src/test/java/com/google/googleinterns/gscribe/services/ExamServiceTest.java

```diff
@@ -0,0 +1,431 @@
+
+            when(spreadsheetService.parseSpreadsheetRequest(user, spreadsheetId, sheetName + "!A:A")).thenReturn(valueRangeForNumberOfQuestions);
+            when(spreadsheetService.parseSpreadsheetRequest(user, spreadsheetId, sheetName + "!A1:G4")).thenReturn(valueRangeWhole);
+
+            examService.getExam(examRequest, user);
+        });
+        assertEquals("Exam format Exception: duration not present in B1", exception.getMessage());
+
+    }
+
+    @Test
+    public void EmptyExamTest() {
+
+        ExamFormatException exception = assertThrows(ExamFormatException.class, () -> {
+            List<List<Object>> valueRangeDataWhole = globalValueRangeDataWholeCopy();
+
+            List<List<Object>> valueRangeDataColumnA = new ArrayList<>();
+            valueRangeDataColumnA.add(Arrays.asList(new String[]{"Duration of exam:"}));
+            valueRangeDataColumnA.add(Arrays.asList(new String[]{"Question Type"}));
+            ValueRange valueRangeColumnA = new ValueRange();
+            valueRangeColumnA.setValues(valueRangeDataColumnA);
+
+            valueRangeDataWhole.remove(3);
+            valueRangeDataWhole.remove(2);
+            valueRangeWhole.setValues(valueRangeDataWhole);
+
+            when(spreadsheetService.parseSpreadsheetRequest(user, spreadsheetId, sheetName + "!A:A")).thenReturn(valueRangeColumnA);
+            when(spreadsheetService.parseSpreadsheetRequest(user, spreadsheetId, sheetName + "!A1:G2")).thenReturn(valueRangeWhole);
+
+            examService.getExam(examRequest, user);
+        });
+        assertEquals("Exam format Exception: Improper exam template used", exception.getMessage());
+
+    }
+
+    @Test
+    public void wrongDurationFormatTest() {
+
+        ExamFormatException exception = assertThrows(ExamFormatException.class, () -> {
+            List<List<Object>> valueRangeDataWhole = globalValueRangeDataWholeCopy();
+            valueRangeDataWhole.get(0).set(1, "asd");
+            valueRangeWhole.setValues(valueRangeDataWhole);
+
+            when(spreadsheetService.parseSpreadsheetRequest(user, spreadsheetId, sheetName + "!A:A")).thenReturn(valueRangeForNumberOfQuestions);
+            when(spreadsheetService.parseSpreadsheetRequest(user, spreadsheetId, sheetName + "!A1:G4")).thenReturn(valueRangeWhole);
+
+            examService.getExam(examRequest, user);
+        });
+        assertEquals("Exam format Exception: duration not in a proper format in B1", exception.getMessage());
+
+    }
+
+    @Test
+    public void durationInNegativeTest() {
+
+        ExamFormatException exception = assertThrows(ExamFormatException.class, () -> {
+            List<List<Object>> valueRangeDataWhole = globalValueRangeDataWholeCopy();
+            valueRangeDataWhole.get(0).set(1, "-10");
+            valueRangeWhole.setValues(valueRangeDataWhole);
+
+            when(spreadsheetService.parseSpreadsheetRequest(user, spreadsheetId, sheetName + "!A:A")).thenReturn(valueRangeForNumberOfQuestions);
```

Comment:
These two lines are there in each and every function.

Reply:
These two lines set the newly made exam objects to be returned to the service call. Due to the change in exam object in every test. These calls need to be after those changes within the test only.